src/braket/circuits/circuit.py

```diff
@@ -436,26 +438,31 @@ def _check_if_qubit_measured(
                 qubit mappings to apply to the `instruction.target`. Key is the qubit in
                 `instruction.target` and the value is what the key will be changed to.
                 Default = `None`.
+            target_qubits (QubitSet | None): the target qubits for a Noise `instruction`.
+                Default = `None`.
 
         Raises:
             ValueError: If adding a gate or noise operation after a measure instruction.
         """
-        if (
-            # check if there is a measure instruction on the target qubit
-            target
-            and target in self._measure_targets
-            # check if there is a measure instruction on any qubits in the target_mapping
-            or (target_mapping and any(targ in self._measure_targets for targ in target_mapping))
-            # If no target or target_mapping is supplied, check if there is a measure
-            # instruction on the current instructions target qubit
-            or (
-                instruction.target
-                and any(targ in self._measure_targets for targ in instruction.target)
+        if self._measure_targets:
+            measure_on_target_qubits = target_qubits and any(target_qubits in self._measure_targets)
+            measure_on_target = target and target in self._measure_targets
```

Comment:
If we use only `target`, then this can modified for both types included in `QubitSetInput`

Reply:
Adjusted this check to account for both types in `QubitSetInput`